warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/ShardReindexJob.java

```diff
@@ -0,0 +1,493 @@
+
+    @Override
+    public int run(String[] args) throws Exception {
+        // parse command line options
+        JCommander cmd = JCommander.newBuilder().addObject(jobConfig).build();
+        cmd.parse(args);
+
+        log.setLevel(Level.DEBUG);
+        Logger.getRootLogger().setLevel(Level.DEBUG);
+
+        Job j = setupJob();
+
+        if (j.waitForCompletion(true)) {
+            return 0;
+        }
+
+        // job failed
+        return -1;
+    }
+
+    private Job setupJob() throws IOException, ParseException, AccumuloException, TableNotFoundException, AccumuloSecurityException, URISyntaxException {
+        configuration.setBoolean("job.cleanupShard", jobConfig.cleanupShard);
+        AccumuloClient.ConnectionOptions<Properties> builder = Accumulo.newClientProperties().to(jobConfig.instance, jobConfig.zookeepers)
+                        .as(jobConfig.username, getPassword());
+
+        // using a batch scanner will force only a single thread per tablet
+        // see AccumuloRecordReader.initialize()
+        if (jobConfig.queryThreads != -1) {
+            builder.batchScannerQueryThreads(jobConfig.queryThreads);
+        }
+
+        // this will not be applied to the scanner, see AccumuloRecordReader.initialize/ScannerImpl
+        if (jobConfig.batchSize != -1) {
+            builder.scannerBatchSize(jobConfig.batchSize);
+        }
+
+        // add resources to the config
+        if (jobConfig.resources != null) {
+            log.info("adding resources");
+            String[] resources = StringUtils.trimAndRemoveEmptyStrings(jobConfig.resources.split("\\s*,\\s*"));
+            for (String resource : resources) {
+                log.info("added resource:" + resource);
+                configuration.addResource(resource);
+            }
+        }
+
+        // setup the accumulo helper
+        AccumuloHelper.setInstanceName(configuration, jobConfig.instance);
+        AccumuloHelper.setPassword(configuration, getPassword().getBytes());
+        AccumuloHelper.setUsername(configuration, jobConfig.username);
+        AccumuloHelper.setZooKeepers(configuration, jobConfig.zookeepers);
+
+        // set the work dir
+        configuration.set(SPLIT_WORK_DIR, jobConfig.workDir);
+        // required for MultiTableRangePartitioner
+        configuration.set("ingest.work.dir.qualified", FileSystem.get(new URI(jobConfig.sourceHdfs), configuration).getUri().toString() + jobConfig.workDir);
+        configuration.set("output.fs.uri", FileSystem.get(new URI(jobConfig.destHdfs), configuration).getUri().toString());
+
+        // setup and cache tables from config
+        Set<String> tableNames = IngestJob.setupAndCacheTables(configuration, false);
+        configuration.setInt("splits.num.reduce", jobConfig.reducers);
+
+        // these are required for the partitioner
+        // split.work.dir must be set or this won't work
+        // job.output.table.names must be set or this won't work
```

Comment:
Does this mean the job will execute but not do the correct operation? Or will the job fail with an illegal arg exception?